docs/DeveloperGuide.adoc

```diff
@@ -767,16 +821,255 @@ Use case ends.
+
+**MSS**
+
+1.  User requests to list all people
+2.  AddressBook shows a list of people
+3.  User requests to tag multiple people
+4.  AddressBook tags all specified people
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. One or more given indexes are invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: List all the persons
+
+*MSS*
+
+1.  User requests to list all the persons
+2.  AddressBook shows a list of persons
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Make relation links with contacts
+
+*MSS*
+
+1.  User requests for name of person to be linked, the name of person with the link, and the relation between both person
+2.   AddressBook displays the details of the person with the link.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The list is empty.
++
+Use case ends.
+
+* 1b. The given name(s) is invalid
++
+Use case ends.
+
+[discrete]
+=== Use case: Sort person by name
+
+*MSS*
+
+1.  User requests to sort person by name
+2.  AddressBook sorts all the contacts
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Sort person by fields in alphabetical order
+
+*MSS*
+
+1.  User requests to sort persons by field
+2.  AddressBook requests user for the field
+3.  User requests to sort persons by a specified field
+4.  AddressBook shows a list of persons by the fields in alphabetical order
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The given field is invalid
++
+Use case resumes at step 2.
+
+* 4a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Sort person to user preference
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to list contacts in new order of preference
+4.  AddressBook shows a list of persons by user preference
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index(es) is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 3b. The given index(es) is repetitive.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Find contacts by tags
+
+*MSS*
+
+1.  User requests to find all the tags
+2.  AddressBook shows a list of tags
+3.  User requests to list persons with specified tag
+4.  AddressBook shows a list of persons with the tag
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no tags.
++
+Use case ends.
+
+* 3a. The given tag is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Running Executable file
+
+*MSS*
+
+1.  User runs the executable file
+2.  AddressBook runs the software
++
+Use case ends.
+
```

Comment:
Overall, the `MSS` interactions are correct and you have written many use cases. Once you update the user stories, don't forget to review the use cases too. Also, pay attention to the numbering in the extensions.